subsys/portability/cmsis_rtos_v2/thread.c

```diff
@@ -509,6 +509,9 @@ __NO_RETURN void osThreadExit(void)
 	__ASSERT(!k_is_in_isr(), "");
 	tid = osThreadGetId();
 
+#if CONFIG_CMSIS_V2_THREAD_DYNAMIC_MAX_COUNT
+	atomic_dec((atomic_t *)&thread_num_dynamic);
+#endif
```

Comment:
This code decrements `thread_num_dynamic` even if the current thread is not a dynamic thread. I assume, this is not what we expect.